src/rust/engine/process_execution/src/local.rs

```diff
@@ -135,8 +149,16 @@ impl super::CommandRunner for CommandRunner {
             )
           })
           // Force workdir not to get dropped until after we've ingested the outputs
-          .map(|result| (result, workdir))
-          .map(|(result, _workdir)| result)
+          .map(move |result| (result, workdir, cleanup_local_dirs) )
```

Comment:
You shouldn't need this change to the map, or to the closure below's arguments; `cleanup_local_dirs` should implicitly be moved into the closure below from the enclosing scope if you refer to it in the body of that closure.

Reply:
ah, indeed - some small journey of compilation errors lead me down that path.

seems like what I ultimately needed there was just the `move` bit up the chain tho, as without that I get:

```
error[E0373]: closure may outlive the current function, but it borrows `cleanup_local_dirs`, which is owned by the current function
   --> process_execution/src/local.rs:124:17
    |
124 |       .and_then(|(output, workdir)| {
    |                 ^^^^^^^^^^^^^^^^^^^ may outlive borrowed value `cleanup_local_dirs`
...
154 |             if !cleanup_local_dirs {
    |                 ------------------ `cleanup_local_dirs` is borrowed here
help: to force the closure to take ownership of `cleanup_local_dirs` (and any other referenced variables), use the `move` keyword
    |
124 |       .and_then(move |(output, workdir)| {
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^
```